src/devices/vdml_imu.c

```diff
@@ -39,9 +41,20 @@ int32_t imu_reset(uint8_t port) {
 	claim_port_i(port - 1, E_DEVICE_IMU);
 	ERROR_IMU_STILL_CALIBRATING(port, device, PROS_ERR);
 	vexDeviceImuReset(device->device_info);
-	// delay for vexos to set calibration flag
-	while(!(vexDeviceImuStatusGet(device->device_info) & E_IMU_STATUS_CALIBRATING)) delay(5);
-	return_port(port - 1, 1);
+	// delay for vexos to set calibration flag, background processing must be called for flag
+	// to be set.
+	uint16_t timeoutCount = 0;
+	// releasing mutex so vexBackgrounProcessing can run without being blocked.
+	port_mutex_give(port);
+	do {
+		delay(5);
+		timeoutCount += 5;
+	} while(!(vexDeviceImuStatusGet(device->device_info) & E_IMU_STATUS_CALIBRATING) && timeoutCount < IMU_RESET_TIMEOUT);
+	if (delay >= IMU_RESET_TIMEOUT) {
```

Comment:
```suggestion
	do {
		port_mutex_give(port - 1);
		delay(5);
		timeoutCount += 5;
		claim_port_i(port - 1, E_DEVICE_IMU);
	} while(!(vexDeviceImuStatusGet(device->device_info) & E_IMU_STATUS_CALIBRATING) && timeoutCount < IMU_RESET_TIMEOUT);
	port_mutex_give(port - 1);
	if (delay >= IMU_RESET_TIMEOUT) {
```
It's probably fine as written, but I don't particularly want to find out what the consequences of calling into vexOS without being under a mutex are since we don't do it anywhere else.